### ISSUE TITLE
Exception on old version of Postgres: column 'attgenerated' does not exist

### ISSUE THREAD

**Issue by duyet** [ClickHouse/ClickHouse#58367]
**Describe what's wrong**

Getting errors while reading from Postgres

```
Code: 36. DB::Exception: Error: ERROR:  column "attgenerated" does not exist
LINE 1: ... atttypid as type_id, atttypmod as type_modifier, attgenerat...
                                                             ^
 (in query: SELECT attname AS name, format_type(atttypid, atttypmod) AS type, attnotnull AS not_null, attndims AS dims, atttypid as type_id, atttypmod as type_modifier, attgenerated as generated FROM pg_attribute WHERE attrelid = (SELECT oid FROM pg_class WHERE relname = '...' AND relnamespace = (SELECT oid FROM pg_namespace WHERE nspname = '...')) AND NOT attisdropped AND attnum > 0 ORDER BY attnum ASC). (BAD_ARGUMENTS) (version 23.12.1.1368 (official build))
```

The recently released version supports retrieving GENERATED and DEFAULT columns (https://github.com/ClickHouse/ClickHouse/commit/806061642a4e45bae2d733d27b6c6482156f7360) from Postgres. It would be nice if this could also support older Postgres servers, as the 'attgenerated' column is not available.

**Does it reproduce on recent release?**

23.12.1.1368

**How to reproduce**

* ClickHouse server: 23.12.1.1368
* Postgres 9
* Running the query below

```sql
SELECT * 
FROM postgresql('<host>:5432', '<database>', '<table>', '<user>', '<pass>')
```

**Expected behavior**

The query should be succeeded.

**Comment by Algunenano** [comment]
It seems all supported Postgresql releases ([currently 12+](https://www.postgresql.org/support/versioning/)) has [this attribute](https://www.postgresql.org/docs/12/catalog-pg-attribute.html). While it'd be nice if somebody added support, it's not something we should focus on.

**Comment by oleg-savko** [comment]
I think should be some flag to support old behaviour or something like think. 
In another case you cant update clickhouse to new version in some legacy application.

**Comment by Algunenano** [comment]
Expecting CH to support unsupported releases of other databases is unrealistic. You can ask Postgres to support `attgenerated` on older versions

**Comment by Artemii-Kravtsov-Second** [comment]
That how it is presented is release logs
```
IMPROVEMENTS 
....
Support PostgreSQL generated columns and default column values in MaterializedPostgreSQL (experimental feature). Closes #40449. #57568
```
Dont you think it `Backward Incompatible Change` rather than an `Improvement`? What used to work - doesn't work anymore. Also Greenplum 6 is based on PostgreSQL 9.4.26

**Comment by igorvoltaic** [comment]
This does affect Greenplum
